priority-queue/PriorityQueue.Shared/QueueManager.cs

```diff
@@ -26,98 +28,101 @@ public QueueManager(string serviceBusConnectionString, string topicName)
         }
 
-        public void ReceiveMessages(string subscription, Func<BrokeredMessage, Task> processMessageTask)
+        public void ReceiveMessages(Func<ServiceBusReceivedMessage, Task> processMessageTask)
         {
-            var options = new OnMessageOptions();
-            options.AutoComplete = true;
+            var options = new ServiceBusProcessorOptions();
+            options.AutoCompleteMessages = true;
             options.MaxConcurrentCalls = 10;
-            options.ExceptionReceived += this.OptionsOnExceptionReceived;
 
-            this.subscriptionClient.OnMessageAsync(
-                 async msg =>
+            this.processor = this.topicClient.CreateProcessor(this.topicName, options);
+
+            processor.ProcessMessageAsync +=
+                 async args =>
                  {
+                     ServiceBusReceivedMessage message = args.Message;
+
                      // Will block the current thread if Stop is called.
                      this.pauseProcessingEvent.WaitOne();
 
                      // Execute processing task here
-                     await processMessageTask(msg);
-                 },
-                 options);
+                     await processMessageTask(message);
+                 };
+            processor.ProcessErrorAsync += this.OptionsOnExceptionReceived;
+
+            processor.StartProcessingAsync();
         }
 
-        public void Setup(string subscription, string priority)
+        public async Task SetupAsync(string subscription, string priority)
         {
-            var namespaceManager = NamespaceManager.CreateFromConnectionString(this.serviceBusConnectionString);
+            var adminClient = new ServiceBusAdministrationClient(this.serviceBusConnectionString);
 
             // Setup the topic.
-            if (!namespaceManager.TopicExists(this.topicName))
+            if (!await adminClient.TopicExistsAsync(this.topicName))
             {
                 try
                 {
-                    namespaceManager.CreateTopic(this.topicName);
+                    await adminClient.CreateTopicAsync(this.topicName);
                 }
-                catch (MessagingEntityAlreadyExistsException)
+                catch (ServiceBusException ex) when (ex.Reason == ServiceBusFailureReason.MessagingEntityAlreadyExists)
                 {
                     Trace.TraceInformation("Messaging entity already created: " + this.topicName);
                 }
                 // It's likely the conflicting operation being performed by the service bus is another queue create operation
                 // If we don't have a web response with status code 'Conflict' it's another exception
-                catch (MessagingException ex) when (((ex.InnerException as WebException)?.Response as HttpWebResponse)?.StatusCode == HttpStatusCode.Conflict)
+                catch (ServiceBusException ex) when (((ex.InnerException as RequestFailedException)?.Status) == 409)
```

Comment:
I suggest to add a constant instead of the hard coded number

Reply:
Updated, thanks a lot!